src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -75,6 +78,77 @@ public class BlobFileStore : IFileStore
         }
     }
 
+    /// <inheritdoc />
+    public async Task<Uri> StoreFileInBlocksAsync(
+        long version,
+        Stream stream,
+        IDictionary<string, long> blockLengths,
+        CancellationToken cancellationToken)
+    {
+        EnsureArg.IsNotNull(stream, nameof(stream));
+        EnsureArg.IsNotNull(blockLengths, nameof(blockLengths));
+        EnsureArg.IsGte(blockLengths.Count, 0, nameof(blockLengths.Count));
+
+        BlockBlobClient blobClient = GetInstanceBlockBlobClient(version);
+
+        int maxBufferSize = (int)blockLengths.Max(x => x.Value);
+
+        await ExecuteAsync(async () =>
+        {
+            byte[] buffer = ArrayPool<byte>.Shared.Rent(maxBufferSize);
+            try
+            {
+                foreach ((string blockId, long blockSize) in blockLengths)
+                {
+#pragma warning disable CA1835 // Prefer the 'Memory'-based overloads for 'ReadAsync' and 'WriteAsync'
```

Comment:
You can also update the `.globalconfig` file as presumably we'll use these APIs if we have the appropriate type:

```
dotnet_diagnostic.CA1835.severity = none
```

Reply:
Will do it in my next PR